R/article.R

```diff
@@ -556,10 +556,69 @@ springer_article <- function(..., keep_tex = TRUE,  citation_package = "natbib",
 #' \samp{https://www.tandf.co.uk/journals/authors/InteractCADLaTeX.zip}.
 #' @export
 #' @rdname article
-tf_article <- function(..., keep_tex = TRUE, citation_package = "natbib") {
+tf_article <- function(..., keep_tex = TRUE, citation_package = "natbib",
+                       biblio_style = "CAD", pandoc_args = NULL) {
+  styles <- list(
+    APA = list(
+      bst = "apacite",
+      cmd = "\\usepackage[natbibapa]{apacite}
+\\setlength\\bibhang{12pt}
+\\renewcommand\\bibliographytypesize{\\fontsize{10}{12}\\selectfont}"
+    ),
+    CAD = list(
+      bst = "tfcad",
+      cmd = "\\usepackage{natbib}
+\\bibpunct[, ]{(}{)}{;}{a}{}{,}"
+    ),
+    NLM = list(
+      bst = "tfnlm",
+      cmd = "\\usepackage[numbers,sort&compress]{natbib}
+\\makeatletter
+\\def\\NAT@def@citea{\\def\\@citea{\\NAT@separator}}
+\\makeatother
+\\bibpunct[, ]{[}{]}{,}{n}{,}{,}
+\\renewcommand\\bibfont{\\fontsize{10}{12}\\selectfont}"
+    ),
+    TFP = list(
+      bst = "tfp",
+      cmd = "\\usepackage[numbers,sort&compress,merge]{natbib}
+\\bibpunct[, ]{(}{)}{,}{n}{,}{,}
+\\renewcommand\\bibfont{\\fontsize{10}{12}\\selectfont}
+\\renewcommand\\citenumfont[1]{\\textit{#1}}
+\\renewcommand\\bibnumfmt[1]{(#1)}"
+    ),
+    TFQ = list(
+      bst = "tfq",
+      cmd = "\\usepackage[numbers,sort&compress]{natbib}
+\\bibpunct[, ]{[}{]}{,}{n}{,}{,}
+\\renewcommand\\bibfont{\\fontsize{10}{12}\\selectfont}"
+    ),
+    TFS = list(
+      bst = "tfs",
+      cmd = "\\usepackage[numbers,sort&compress]{natbib}
+\\bibpunct[, ]{[}{]}{,}{n}{,}{,}
+\\renewcommand\\bibfont{\\fontsize{10}{12}\\selectfont}"
+    )
+  )
+  if (! biblio_style %in% names(styles))
+    stop(
+      paste(
+        "Invalid biblio_style in Taylor and Francis article. Allowed values are:",
+        paste(names(styles), collapse = ", ")
+      )
+    )
+  sty <- styles[[biblio_style]]
+  pandoc_args <- c(
+    pandoc_args,
+    rmarkdown::pandoc_variable_arg("bst-name", sty$bst),
+    rmarkdown::pandoc_variable_arg("biblio-commands", sty$cmd)
+  )
```

Comment:
Really good idea to use this pandoc variable to fill the template accordingly ! 

a `switch(biblio_style, )` could also be use to make it clear this is a choice between a set of possibly value. though it does not probably improve much except readibility maybe